src/components/dashboard/main-content/course-enrollments/data/hooks.js

```diff
@@ -111,97 +117,133 @@ export const useCourseEnrollments = ({
 
 /**
  * Return data for upgrading a course using the user's subsidies
+ * Based on the applicable subsidy, the upgrade URL, applicable subsidy metadata and toggled loading state
+ * will return in the following prioritization order:
+ *  - Subscriptions
+ *  - Coupon codes
+ *  - Learner Credit Subsidy
+ *
  * @param {object} args Arguments.
  * @param {String} args.courseRunKey id of the course run
- * @param {String} args.enterpriseId UUID of the enterprise
- * @param {Object} args.subscriptionLicense the user's subscription license
- * @param {Object} args.location location object from useLocation()
- *
- * @returns {Object} { isLoading, licenseUpgradeUrl, couponUpgradeUrl, upgradeUrl }
+ * @param {String} args.mode The mode of the course. Used as a gating mechanism for upgradability
+ * default: false
+ * @returns {Object} {
+ *     licenseUpgradeUrl: undefined,
+ *     couponUpgradeUrl: undefined,
+ *     learnerCreditUpgradeUrl: undefined,
+ *     subsidyForCourse: undefined,
+ *     courseRunPrice: undefined,
+ *     }
  */
 export const useCourseUpgradeData = ({
   courseRunKey,
-  enterpriseId,
-  subscriptionLicense,
-  couponCodes,
-  location,
+  mode,
 }) => {
-  const [licenseUpgradeUrl, setLicenseUpgradeUrl] = useState();
-  const [couponUpgradeUrl, setCouponUpgradeUrl] = useState();
-  const [subsidyForCourse, setSubsidyForCourse] = useState();
-  const [courseRunPrice, setCourseRunPrice] = useState();
+  const location = useLocation();
+  // We determine whether the course mode is such that it can be upgraded
+  const canUpgradeToVerifiedEnrollment = [COURSE_MODES_MAP.AUDIT, COURSE_MODES_MAP.HONOR].includes(mode);
+  const { authenticatedUser } = useContext(AppContext);
+  const { data: enterpriseCustomer } = useEnterpriseCustomer();
+  const { data: customerContainsContent } = useEnterpriseCustomerContainsContent([courseRunKey]);
 
-  const [isLoading, setIsLoading] = useState(true);
+  // TODO: Remove authenticatedUser?.administrator flag when rolling out
```

Comment:
If there's a ticket number for executing the rollout, I'd recommend putting it in this comment.

Reply:
Yep there was a quick ticket written up on it which I will reference here 👍🏽